interfaces/inc/sync.h

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef SYNC_H
+#define SYNC_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef _WIN32
+#ifndef volatile_atomic
+#define volatile_atomic volatile
+#else
+#define volatile_atomic volatile _Atomic
+#endif
+#endif

```

Comment:
Probably want to just include interlocked.h here or put the volatile_atomic define in its own header.